File: Java/프로그래머스/2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.java
class Solution {
    public int solution(int n, int k) {
        int answer = 0;
        String number = Integer.toString(n, k);
        String[] arr = number.split("0");

        for (String s : arr) {
            if (s.length() == 0) {
                continue;
            }
            if (isPrime(Long.parseLong(s))) {
                answer++;
            }
        }

        return answer;
    }
    
    boolean isPrime(long n) {
        if (n < 2) {
            return false;
        }

        for (long i = 2; i * i <= n; i++) {
            if (n % i == 0) {
                return false;
            }
        }

        return true;
    }
}

File: Python/프로그래머스/2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.py
def solution(n, k):
    answer = 0
    arr = base(n, k).split("0")
    for num in arr:
        if num and isPrime(int(num)):
            answer += 1
    return answer


def isPrime(n):
    if n < 2:
        return False

    for i in range(2, int(n**0.5) + 1):
        if n % i == 0:
            return False

    return True


def base(n, q):
    result = ""
    while n > 0:
        n, remainder = divmod(n, q)
        result += str(remainder)
    return result[::-1]